lru.go

```diff
@@ -43,6 +48,21 @@ func NewWithEvict[K comparable, V any](size int, onEvicted func(key K, value V))
 	return
 }
 
+// NewSieveWithEvict constructs a fixed size cache with the given eviction
+// callback.
+func NewSieveWithEvict[K comparable, V any](size int, onEvicted func(key K, value V)) (c *Cache[K, V], err error) {
+	// create a cache with default settings
+	c = &Cache[K, V]{
+		onEvictedCB: onEvicted,
+	}
+	if onEvicted != nil {
+		c.initEvictBuffers()
+		onEvicted = c.onEvicted
+	}
+	c.lru, err = simplelru.NewSieve(size, onEvicted)
```

Comment:
if they are ok, I think we can change `c.lru` to `c.cache`, minor though

Reply:
Done!

Reply:
awesome